com.unity.multiplayer.mlapi/Runtime/Spawning/NetworkPrefabHandler.cs

```diff
@@ -40,7 +40,8 @@ public interface INetworkPrefabInstanceHandler
         /// The most common approach is to make the <see cref="NetworkObject"/> inactive by calling <see cref="GameObject.SetActive(bool)"/>.
         /// </summary>
         /// <param name="networkObject">The <see cref="NetworkObject"/> being destroyed</param>
-        void Destroy(NetworkObject networkObject);
+        /// <returns>(true) destroy the <see cref="GameObject"/> (false) do not destroy the <see cref="GameObject"/> </returns>
```

Comment:
Just wondering why this is needed? Couldn't the handler just destroy the object itself if it wanted?

Reply:
Nope, you end up with a scenario where you are pooling NetworkObjects and then detect that you want to say...unload a scene and the server side starts despawning everything that needs to be destroyed.  If you don't have the ability to be able to allow that condition, then you run into the issue of "how does a client really allow a pooled object with handler to be destroyed"?  This allows the user to specify whether they want the normal "despawn or destroy" to actually destroy the NetworkObject or not (and solves several edge case scenarios).